providers/fips/fipsprov.c

```diff
@@ -342,22 +371,27 @@ static const OSSL_ALGORITHM fips_ciphers[] = {
     { "AES-128-CTR", "fips=yes", aes128ctr_functions },
     { "AES-256-XTS", "fips=yes", aes256xts_functions },
     { "AES-128-XTS", "fips=yes", aes128xts_functions },
-    /* TODO(3.0) Add aliases for these ciphers */
-    { "id-aes256-GCM", "fips=yes", aes256gcm_functions },
-    { "id-aes192-GCM", "fips=yes", aes192gcm_functions },
-    { "id-aes128-GCM", "fips=yes", aes128gcm_functions },
-    { "id-aes256-CCM", "fips=yes", aes256ccm_functions },
-    { "id-aes192-CCM", "fips=yes", aes192ccm_functions },
-    { "id-aes128-CCM", "fips=yes", aes128ccm_functions },
-    { "id-aes256-wrap", "fips=yes", aes256wrap_functions },
-    { "id-aes192-wrap", "fips=yes", aes192wrap_functions },
-    { "id-aes128-wrap", "fips=yes", aes128wrap_functions },
-    { "id-aes256-wrap-pad", "fips=yes", aes256wrappad_functions },
-    { "id-aes192-wrap-pad", "fips=yes", aes192wrappad_functions },
-    { "id-aes128-wrap-pad", "fips=yes", aes128wrappad_functions },
+    { "AES-256-GCM:id-aes256-GCM", "fips=yes", aes256gcm_functions },
+    { "AES-192-GCM:id-aes192-GCM", "fips=yes", aes192gcm_functions },
+    { "AES-128-GCM:id-aes128-GCM", "fips=yes", aes128gcm_functions },
+    { "AES-256-CCM:id-aes256-CCM", "fips=yes", aes256ccm_functions },
+    { "AES-192-CCM:id-aes192-CCM", "fips=yes", aes192ccm_functions },
+    { "AES-128-CCM:id-aes128-CCM", "fips=yes", aes128ccm_functions },
+    { "AES-256-WRAP:id-aes256-wrap:AES256-WRAP", "fips=yes",
+      aes256wrap_functions },
+    { "AES-192-WRAP:id-aes192-wrap:AES192-WRAP", "fips=yes",
+      aes192wrap_functions },
+    { "AES-128-WRAP:id-aes128-wrap:AES128-WRAP", "fips=yes",
+      aes128wrap_functions },
+    { "AES-256-WRAP-PADid-aes256-wrap-pad::AES256-WRAP-PAD", "fips=yes",
```

Comment:
oops.. missing seperator ':' which is why some travis tests are failing in fips.
Still not sure that I like these strings being duplicated across providers.
Would defines be any better?

Reply:
following 2 strings are also missing the :

Reply:
#define OSSL_CIPHER_NAME_AES_256_WRAP_PAD ""AES-256-WRAP-PAD:id-aes256-wrap-pad"

Reply:
I kind of think that the FIPS provider could get away with just defining the names NIST uses.

With #10070 being a reasonable prospect, we'll be reconsidering how this works...

Reply:
My original proposal was to have symbols that carried the names, alongside the implementations, and having fipsprov.c, defltprov.c et al simply use those symbols.

#10070 certainly changes things, but not so much the central naming mechanism, so it shouldn't affect this PR.

Reply:
what should ChaCha20-Poly1305 look like?

Reply:
Its a combination of two algorithms. According to the naming scheme:

ALGNAME[VERSION?][-SUBNAME[VERSION?]?][-SIZE?][-MODE?]

The first algname is CHACHA20, and there is a subname POLY1305. Therefore:

CHACAH20-POLY1305